README.md

```diff
@@ -57,113 +69,158 @@ Subdirectory README files should follow the same general format as this root fil
+
+<hr>
+
+## ENDPOINTS
+
+All endpoints (Tier 1 and Tier 2) adhere to RESTful standards and communicate using JSON. 
+
+All endpoints are served at the URL in the following format:
+
+https://opencap.{domain}.{extension}:443/{endpoint}
+
+HTTPS is required for all CAP endpoints, and as is standard the port 443 will be used.
+
+Endpoints are prefixed with Protocol version in case of non-backwards compatibility.
+
+<hr>
+
+## TIER 1 ENDPOINT
 
-### GET /address/{username}/{coin}
+The following GET endpoint is the only endpoint a server needs to support to be CAP 1 compliant. This is because in order for a user to have a CAP alias, wallets just need to know where to look up the address. CAP 1 can be as simple as a static string served from the endpoint.
+
+### GET /v1/address/{ledger_id}/{username}
 
 No Authorization (Public method)
 
-Returns the address of the related username and coin, if they exist.
+Returns the address of the related username and ledger, if it exists on the server.
 
 #### Example Usage
 
 * Bob's decides to send Alice 5 BTC via her alias alice@ogdolo.com
 * Bob's wallet parses the alias into a URL destination and HTTP method
-* alice@ogdolo.com -> GET <https://cap.ogdolo.com/address/alice/btc>
+* BTC_SEGWIT = ledger_id: 1
+* alice@ogdolo.com -> GET <https://opencap.ogdolo.com/address/1/alice>
 * If alice truly has a Bitcoin address hosted on the domain that her alias suggests, the address will be sent back to Bob's wallet with an HTTP 200 Status OK.
 
 ```javascript
 Code: 200
 {
-    "address": "bc1qvw0ytfntx6zs0lfsruem6xwj0mewng523ktatp",
+	"message": "success",
+	"data": {
+        "address_type": 0,
+		"address": "bc1qvw0ytfntx6zs0lfsruem6xwj0mewng523ktatp",
+		"extensions": {
+			"dnssig": "..."
+		}
+	}
 }
 ```
 
 * Bob's wallet can notify Bob that a valid address was found, and the wallet can now route a payment to Alice through her address.
 
-### POST /auth
+A wallet that uses multiple "ledgers" (BTC has legacy addresses and segwit addresses that constitute different ledgers within CAP) would first query the endpoint using the ledger_id of the preferred ledger, and if that fails it can try other ledgers. If the sender was able to send to a segwit address, they would first query the BTC_SEGWIT id (1) and if that fails, they would try the BTC_LEGACY id (0).
 
-No Authorization (Public method)
+The "address_type"
 
-Gets a new JWT for the given user. The JWT can be used to authenticate with the other endpoints. JWT should last 5 minutes.
+<hr>
+
+## TIER 2 ENDPOINTS
+
+The following endpoints are required for a server to be CAP 2 compliant. These endpoints allow a wallet to know how to interact with a server on behalf of the user, in other words CREATE/UPDATE/DELETE address/user info.
+
+### POST /v1/address

```

Comment:
`PATCH /v1/users/{username}/ledgers/{ledger_id}` following the style from above.

Reply:
When doing a post its better to have the params sent as JSON I think. Thats the convention.

Reply:
[Apparently, PATCH is used when only once field gets updated and PUT for the whole entitity](https://stackoverflow.com/questions/28459418/rest-api-put-vs-patch-with-real-life-examples). We should use PUT as POST implies that it fails with 409 if the resource already exists.